### PR TITLE
Implementa dark mode com base nos tokens do Design System

### LINKED ISSUE #88: Implementar dark mode com base nos tokens do Design System
## Descrição

O projeto já possui referência visual e tokens para tema escuro nos arquivos de design em `assets/`, especialmente em `assets/tokens.css`, `assets/Carlesso Admin.html`, `assets/Componentes.html` e `assets/Fundacao.html`. No Angular, os tokens equivalentes estão em `src/styles/_tokens.scss` e já existe o `StylePreferencesService`, que aplica `data-theme` e `data-density` no `documentElement`.

Apesar disso, o sistema ainda não oferece uma experiência completa de dark mode para o usuário final: falta uma forma de alternar o tema pela interface, persistir a preferência e validar se todos os componentes/telas usam corretamente os tokens de cor.

## Objetivo

Implementar o dark mode do sistema usando os tokens existentes do Design System, garantindo alternância entre tema claro e escuro, persistência da preferência do usuário e aderência visual às referências em `assets/`.

## Escopo da implementação

- Revisar os tokens de dark mode em `assets/tokens.css` e `src/styles/_tokens.scss`, mantendo os dois arquivos sincronizados quando houver ajuste.
- Usar o atributo `data-theme="dark"` como mecanismo principal de aplicação do tema, conforme definido nos assets.
- Evoluir o `StylePreferencesService` para persistir a preferência de tema, por exemplo em `localStorage`.
- Adicionar controle visual para alternar entre tema claro e escuro em uma área global do sistema, preferencialmente na navbar ou em outro ponto persistente da aplicação.
- Garantir que o estado inicial do tema considere a preferência salva; se não houver preferência, avaliar uso de `prefers-color-scheme` como fallback.
- Ajustar estilos globais e componentes que ainda dependam de cores fixas ou variáveis antigas, priorizando tokens como `--bg-app`, `--bg-elev`, `--bg-paper`, `--text-primary`, `--text-secondary`, `--text-muted`, `--border-*` e cores funcionais.
- Validar telas principais do sistema em dark mode, incluindo login, dashboard/início, listas, formulários, detalhes, relatórios e área administrativa.
- Manter compatibilidade com o tema claro existente.

## Critérios de aceite

- O usuário consegue alternar entre tema claro e escuro pela interface.
- A preferência selecionada permanece após recarregar a página e ao reabrir o sistema no mesmo navegador.
- O atributo `data-theme` é aplicado corretamente no elemento raiz (`html`) com os valores `light` e `dark`.
- O dark mode usa os tokens definidos no Design System, sem introduzir paleta paralela ou cores hardcoded desnecessárias.
- Componentes comuns como navbar, botões, cards, tabelas, formulários, badges, dialogs e mensagens mantêm contraste adequado em tema escuro.
- As telas principais continuam legíveis e utilizáveis em tema claro e escuro.
- O tema claro não sofre regressões visuais relevantes.
- Testes automatizados existentes continuam passando.
- Novos testes cobrem a persistência/aplicação da preferência de tema e, se aplicável, o comportamento do controle de alternância.

## Sugestão de validação/testes

- Executar a suíte automatizada do projeto, por exemplo `npm test` ou o comando padrão configurado no repositório.
- Adicionar/atualizar testes unitários do `StylePreferencesService` para cobrir:
  - carregamento da preferência salva;
  - persistência ao alternar tema;
  - aplicação de `data-theme` no `documentElement`;
  - fallback quando não houver preferência salva.
- Adicionar teste do componente que expõe o controle de tema, validando clique/ação e chamada ao serviço.
- Fazer validação manual em tema claro e escuro nas principais jornadas:
  - login/logout;
  - navegação pela navbar;
  - listas de pacientes/profissionais/usuários;
  - formulários de cadastro/edição;
  - detalhes do paciente;
  - relatórios;
  - diálogos de confirmação.
- Conferir contraste visual de textos, bordas, estados hover/focus, botões desabilitados e mensagens funcionais.

## Observações técnicas

- `assets/README.md` informa que os protótipos em `assets/` não fazem parte do build Angular; eles devem ser usados como referência visual.
- Os tokens do Angular ficam em `src/styles/_tokens.scss`; alterações em tokens devem ser refletidas também em `assets/tokens.css` quando forem mudanças de Design System.
- O serviço `src/app/core/services/style-preferences.service.ts` já aplica `data-theme` e `data-density`, então a implementação pode reaproveitar essa base em vez de criar um mecanismo paralelo.
- Evitar cores fixas em componentes; preferir variáveis CSS existentes.
- Considerar acessibilidade: foco visível, contraste mínimo e labels/aria-label no controle de tema.

## Labels sugeridas

- `enhancement`
- `frontend`
- `design-system`
- `accessibility`

### PR DESCRIPTION
**Pull request by FabioCarlesso**
## Resumo

Implementa o dark mode completo para o usuário final, reutilizando os tokens `[data-theme="dark"]` já definidos no Design System.

- **`StylePreferencesService`**: passa a persistir tema e densidade em `localStorage` (chave `carlesso.style-preferences`); resolve o tema inicial pela preferência salva e, na ausência dela, usa `prefers-color-scheme` como fallback — sem persistir o valor do fallback, para que a preferência do SO continue valendo até o usuário escolher. Novo método `toggleTheme()`; `setTheme`/`setDensity`/`apply` agora gravam no storage.
- **Navbar (`AppComponent`)**: novo botão **Tema claro/Tema escuro** (`.btn-tema`), agrupado com **Sair** em `.navbar-actions`, com `aria-label` e `aria-pressed` para acessibilidade.
- **Tokens**: nenhum ajuste necessário — `src/styles/_tokens.scss` e `assets/tokens.css` já estavam sincronizados e completos para o tema escuro. Sem paleta paralela ou cores hardcoded de tema na camada de componentes.

## Motivo

O projeto já tinha os tokens de tema escuro e o `StylePreferencesService` aplicando `data-theme`, mas faltava a experiência de dark mode para o usuário final: alternância pela interface, persistência da preferência e estado inicial considerando preferência salva / `prefers-color-scheme`. Referência: issue #88.

## Testes executados

- `npx ng test --watch=false --browsers=ChromeHeadless` — 552 testes passando, incluindo:
  - `StylePreferencesService`: carregamento da preferência salva, persistência em `setTheme`/`setDensity`/`toggleTheme`/`apply`, aplicação de `data-theme`/`data-density` no `documentElement`, fallback para `prefers-color-scheme`, preferência corrompida ignorada e fallback não persistido no startup.
  - `AppComponent`: renderização do controle de tema na navbar, ocultação quando não autenticado, chamada a `toggleTheme()` no clique e rótulos/`aria-*` conforme o tema ativo.
- `npx ng build` — build de produção concluído com sucesso.
- Validação visual manual no navegador não executada neste ambiente (sem browser interativo).

## Arquivos principais alterados

- `src/app/core/services/style-preferences.service.ts` — persistência, fallback e `toggleTheme()`
- `src/app/app.component.ts` / `.html` — botão de alternância de tema na navbar
- `src/styles.scss` — estilos de `.btn-tema` e `.navbar-actions`
- specs de `StylePreferencesService` e `AppComponent`
- `docs/context.md` e `README.md` — documentação do dark mode

## Referência

Closes #88

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/context.md [hunk 1]
```diff
@@ -67,7 +68,14 @@ Todos os componentes são standalone (sem NgModules), seguindo a arquitetura mod
...
+
+O dark mode em si é definido inteiramente pelos tokens em `[data-theme="dark"]` (`src/styles/_tokens.scss`, espelhado em `assets/tokens.css`), que sobrescrevem superfícies (`--bg-*`), texto (`--text-*`), bordas (`--border-*`), sombras e fundos funcionais. Não há paleta paralela nem cores hardcoded específicas de tema na camada de componentes.
+
+### Controle de tema na navbar
+A navbar do `AppComponent` expõe o botão **Tema claro/Tema escuro** (`.btn-tema`), agrupado com o botão **Sair** em `.navbar-actions`. O botão chama `StylePreferencesService.toggleTheme()` e expõe `aria-pressed` (estado do tema escuro) e `aria-label` descritivo da ação. O controle só aparece quando o usuário está autenticado, junto com o restante da navbar; a tela de login continua respeitando o tema persistido ou o `prefers-color-scheme`.
```
Comment (FabioCarlesso) [inline docs/context.md]: Problem: This documentation says the theme control only appears when the user is authenticated and that the login screen only respects the persisted/OS theme, but the PR also adds an unauthenticated `.login-theme-toggle` on the login page.

Why it matters: `context.md` is used as project context for future work. As written, it can mislead maintainers into removing or duplicating the login-page control because the documented behavior does not match the implemented UI.

Suggested improvement: Update this section to explicitly document both controls: the navbar toggle for authenticated users and the login-page toggle for unauthenticated users.

Severity: Low

File: src/app/core/services/style-preferences.service.ts [hunk 2]
```diff
@@ -36,10 +39,79 @@ export class StylePreferencesService {
...
+      const theme = parsed.theme === 'light' || parsed.theme === 'dark' ? parsed.theme : null;
+      const density =
+        parsed.density === 'default' || parsed.density === 'compact' || parsed.density === 'comfortable'
+          ? parsed.density
+          : null;
+      if (!theme && !density) {
+        return null;
+      }
+      return {
+        theme: theme ?? DEFAULT_PREFERENCES.theme,
```
Comment (FabioCarlesso) [inline src/app/core/services/style-preferences.service.ts]: Problem: A partial stored value with a valid `density` but missing/invalid `theme` is treated as a saved preference and forces `theme` back to `DEFAULT_PREFERENCES.theme` (`light`), bypassing the `prefers-color-scheme` fallback.

Why it matters: If the storage value is partially corrupted or comes from an older/future schema that only stored density, users whose OS prefers dark mode will still get light mode on startup, contrary to the documented precedence of saved theme first, then OS theme fallback.

Suggested improvement: Resolve theme and density independently. For example, preserve a valid stored density, but when `theme` is absent/invalid, use `this.prefersDarkScheme() ? 'dark' : 'light'` instead of `DEFAULT_PREFERENCES.theme`; add a spec for `{ density: 'compact' }` with `prefers-color-scheme: dark`.

Severity: Low

**Review by FabioCarlesso — COMMENTED** [review]
I reviewed the PR across documentation, bugs/edge cases, security, performance, Angular practices, API integration, UI/UX, tests, and code quality. I left two inline findings: one documentation mismatch and one edge case in stored preference recovery. Security, API integration, performance, and the added unit-test coverage otherwise look good for this scope. Local verification: `npx ng test --watch=false --browsers=ChromeHeadless` passed with 556 specs, and `npx ng build` completed successfully.

**Comment by FabioCarlesso** [comment]
Correções aplicadas no commit `834ade4`:

- Atualizei `README.md` e `docs/context.md` para documentar que o controle de tema existe tanto na navbar autenticada quanto na tela de login.
- Ajustei `StylePreferencesService` para, quando houver preferência salva parcial com densidade válida mas tema ausente/inválido, preservar a densidade e resolver o tema via `prefers-color-scheme` em vez de forçar `light`.
- Adicionei teste cobrindo `{ theme: 'invalid', density: 'compact' }` com preferência do SO em dark.

Verificações executadas:
- `npx ng test --watch=false --browsers=ChromeHeadless` — 557 specs passando.
- `npx ng build` — build concluído com sucesso.
